### PR TITLE
Ruby 2.0 compat

### PR DESCRIPTION
**Pull request by Meatballs1**
iconv is deprecated in Ruby > 2.0. 

Not sure if this works for the FFI::StructLayout::CharArray modifications..?

### CODE REVIEW

**Comment by mydoghasworms** [comment]
Sorry, I've been meaning to come back to you on this one. The deprecation warning has always been bothering me, and I have been meaning to adapt for 2.0, but I was always worried about breaking it for any other users out there. Maybe it's just a good idea to make it work for 2.0 and then users who need it for older versions can use the older gem or fork it. What do you think?

**Comment by Meatballs1** [comment]
I'm happy with any approach, I'm currently pointing the gem to my own github repo but will reset it back when upstream gets updated :)

I have no real idea if my approach is the best way, encoding always scares me...

File: lib/nwrfc/nwrfclib.rb [hunk 1]
```diff
@@ -49,17 +48,17 @@ def read_string_dn(max=0)
...
 # architectures, though the plan is not to support them)
-#String.class_eval{define_method(:cU){ Iconv.conv("UTF-16LE", "UTF8", self+"\0") }}
 class String
 
-  # Convert string from UTF-8 to doudble-null terminated UTF-16LE string
+  # Convert string from UTF-8 to double-null terminated UTF-16LE string
   def cU
-    NWRFCLib::Cutf8_to_utf16le.iconv(self+"\0")
+    duplicate_string = self.dup << "\0"
+    duplicate_string.force_encoding('UTF-8').encode('UTF-16LE')
```
Comment (mydoghasworms) [inline lib/nwrfc/nwrfclib.rb]: What is the reason for duplicating the String here?

**Comment by mydoghasworms** [comment]
I have applied the changes, but kept support for 1.8 and older implementations, so I will not be merging this pull request. Thanks, and sorry it took so long!

**Pull request closed without merging**